pkg/controller/istiocsr/controller.go

```diff
@@ -0,0 +1,341 @@
+package istiocsr
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/selection"
+	"k8s.io/apimachinery/pkg/types"
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/client-go/tools/record"
+
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	"sigs.k8s.io/controller-runtime/pkg/cache"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	"github.com/go-logr/logr"
+
+	certmanagerv1 "github.com/cert-manager/cert-manager/pkg/apis/certmanager/v1"
+
+	v1alpha1 "github.com/openshift/cert-manager-operator/api/operator/v1alpha1"
+)
+
+var (
+	// requestEnqueueLabelKey is the label key name used for filtering reconcile
+	// events to include only the resources created by the controller.
+	requestEnqueueLabelKey = "app"
+
+	// requestEnqueueLabelValue is the label value used for filtering reconcile
+	// events to include only the resources created by the controller.
+	requestEnqueueLabelValue = "cert-manager-istio-csr"
+)
+
+// Reconciler reconciles a IstioCSR object
+type Reconciler struct {
```

Comment:
prefer `IstioCSRReconciler` to be more apt (I understand this was scaffold-ing generated)
The way I see it is that it helps us add new XyzReconciler if we really need to add new resource for IstioCSR later on in the same directory.

Reply:
It was done intentionally to avoid redundancy with the package name. Even when we add new resource, since this is the original/intended resource for the controller, I would like to keep it as just `Reconciler`.  Let me know if it needs to be updated.